### PR TITLE
Undefined Ratios and Fairness Metric Testing

### PR DESCRIPTION
**Pull request by camagallen**
## Overview
AIF360's sklearn.metric.ratio function returns 0.0 where the denominator of a ratio is 0, rather than undefined. This can be misleading, although the other functionality is still useful. This PR adds code to return NaN values in those cases.

Also in this PR are related updates to test__fairnessmetrics to increase robustness. Some cleanup was done in the associated scripts __testing_utilites and _preprocessing, and a ceiling was placed on the version number for AIF360 in setup.py.

Resolves: bugs detected during documentation update

## How was this update tested?
pytest and active testing in ToolUsage notebooks

## Checklist
- [x] I have read the [Contributing](https://github.com/KenSciResearch/docs/code_contributions/CONTRIBUTING.md) guidelines
- [ ] I performed a self-review of my own code
- [x] The CHANGELOG and version.json have been updated
- [x] Documentation has been updated to reflect the changes
    - [x] The code has clear and helpful comments, particularly in hard-to-understand areas
    - [x] READMEs, Tutorials/Examples, and links have been updated as appropriate
- [x] Tests have been added to prove that the fix is effective or that the feature works
- [x] Pytest was run; new and existing unit tests pass locally with my changes
- [x] I have made note of any breakages or dependencies (if applicable)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] These changes generate no new warnings

### CODE REVIEW

**Comment by camagallen** [comment]
Dropping this PR in favor of newer PR (#66)
